File: packages/Admob/ios/Classes/FLTTeadsMediationNetworkExtras.swift
//
//  FLTTeadsMediationNetworkExtras.swift
//  teads_admob_adapter
//
//  Created by Thibaud Saint-Etienne on 18/10/2022.
//

import Flutter
import google_mobile_ads
import TeadsAdMobAdapter
import TeadsSDK

public class FLTTeadsMediationNetworkExtras: FLTMediationNetworkExtrasProvider {
        
    public init() { }
    
    public func getMediationExtras(_ adUnitId: String, mediationExtrasIdentifier: String?) -> [GADAdNetworkExtras]? {
        return [FLTTeadsAdapterSettings.instance.adapterSettings]
    }
    
}

File: packages/TeadsAdmobAdapter/android/src/main/kotlin/tv/teads/teadsadmobadapter/teads_admob_adapter/FLTTeadsMediationNetworkExtras.kt
package tv.teads.teadsadmobadapter.teads_admob_adapter

import android.os.Bundle

import com.google.android.gms.ads.mediation.MediationExtrasReceiver

import io.flutter.plugins.googlemobileads.MediationNetworkExtrasProvider
import tv.teads.adapter.admob.TeadsAdapter

class FLTTeadsMediationNetworkExtras : MediationNetworkExtrasProvider {
    override fun getMediationExtras(adUnitId: String, identifier: String?): Map<Class<out MediationExtrasReceiver?>, Bundle>? {
        val teadsBundle: Bundle = FLTTeadsAdapterSettings.instance.mediationSettingsBuilder.build().toBundle()
        val extras: MutableMap<Class<out MediationExtrasReceiver?>, Bundle> = HashMap()
        extras[TeadsAdapter::class.java] = teadsBundle
        // Note: You can pass extras to multiple adapters by adding more entries.
        return extras
    }
}